### PR TITLE
Parallel accuracy

### PR DESCRIPTION
**Pull request by shak360**

### CODE REVIEW

File: lineage/Analyze.py [hunk 1]
```diff
@@ -70,10 +70,12 @@ def run_Analyze_over(list_of_populations, num_states):
     list_of_populations: a list of populations that contain lineages
     num_states: an integer number of states to identify (a hyper-parameter of our model)
     """
-    output = []
-    for population_idx, population in enumerate(list_of_populations):
-        output.append(Analyze(population, num_states))
-    return output
+    input_args = [(population, num_states) for population in list_of_populations]
```
Comment (shak360) [inline lineage/Analyze.py]: Confused on if this is the right way to do. It's throwing an an error about how `Analyze` is expecting an additional argument of `numStates`, but I wrote in the previous line code to fix this, so I'm not sure...

Comment (shak360) [reply lineage/Analyze.py]:
```
shakthi@aretha:~/Projects/CAPSTONE$ make
mkdir -p ./manuscript/figures
. venv/bin/activate && ./genFigures.py 3
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 198, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 198, in <listcomp>
    return [fn(*args) for args in chunk]
TypeError: Analyze() missing 1 required positional argument: 'num_states'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./genFigures.py", line 14, in <module>
    ff = eval(nameOut + '.makeFigure()')
  File "<string>", line 1, in <module>
  File "/home/shakthi/Projects/CAPSTONE/lineage/figures/figure3.py", line 53, in makeFigure
    accuracies, tr = accuracy_increased_cells()
  File "/home/shakthi/Projects/CAPSTONE/lineage/figures/figure3.py", line 108, in accuracy_increased_cells
    results_holder = run_Results_over(output)
  File "/home/shakthi/Projects/CAPSTONE/lineage/Analyze.py", line 195, in run_Results_over
    res_holder = e.map(Results, output)
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 671, in map
    timeout=timeout)
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 587, in map
    fs = [self.submit(fn, *args) for args in zip(*iterables)]
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 587, in <listcomp>
    fs = [self.submit(fn, *args) for args in zip(*iterables)]
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 184, in _get_chunks
    chunk = tuple(itertools.islice(it, chunksize))
  File "/usr/lib/python3.7/concurrent/futures/process.py", line 483, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 598, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 428, in result
    return self.__get_result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
TypeError: Analyze() missing 1 required positional argument: 'num_states'
make: *** [makefile:18: output/figure3.svg] Error 1
```

**Comment by shak360** [comment]
https://stackoverflow.com/questions/10212445/python-map-list-item-to-function-with-arguments Looks like the second answer on this answers my question:

need to use https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.starmap

**Comment by shak360** [comment]
`ProcessPoolExecutor` doesn't seem to have `starmap`

**Comment by aarmey** [comment]
@shak360 this is good to go as far as the parallel running is concerned. I'm getting an unrelated singular matrix error, though. Let me know if you want me to look into it.

File: lineage/StateDistribution.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def pdf(self, tuple_of_obs):  # user has to define how to calculate the likeliho
         # distribution observations), so the likelihood of observing the multivariate observation is just the product of
         # the individual observation likelihoods.
 
-        bern_ll = sp.bernoulli.pmf(k=tuple_of_obs[0], p=self.bern_p)  # bernoulli likelihood
+        bern_ll = self.bern_p**(tuple_of_obs[0]) * (1.0 - self.bern_p)**(tuple_of_obs[0])
```
Comment (aarmey) [inline lineage/StateDistribution.py]: @shak360    I think this is what's wrong. `tuple_of_obs[0]` is an array, so you're turning `bern_ll` into an array.

**Comment by shak360** [comment]
some more minor errors to fix

File: lineage/StateDistribution.py [hunk 3]
```diff
@@ -31,8 +31,8 @@ def pdf(self, tuple_of_obs):  # user has to define how to calculate the likeliho
         # distribution observations), so the likelihood of observing the multivariate observation is just the product of
         # the individual observation likelihoods.
 
-        bern_ll = sp.bernoulli.pmf(k=tuple_of_obs[0], p=self.bern_p)  # bernoulli likelihood
-        gamma_ll = sp.gamma.pdf(x=tuple_of_obs[1], a=self.gamma_a, scale=self.gamma_scale)  # gamma likelihood
+        bern_ll = self.bern_p**(tuple_of_obs[0]) * (1.0 - self.bern_p)**(tuple_of_obs[0])
```
Comment (aarmey) [inline lineage/StateDistribution.py]: This is wrong. You need to check whether `tuple_of_obs[0]` is 1 or 0. Right now it's the same whether it's 0 or 1.

Comment (shak360) [reply lineage/StateDistribution.py]: My bad, told myself to change this but forgot. I think the second exponent needs to be `1-tuple_of_obs[0]`.

**Comment by shak360** [comment]
This should be done @aarmey . Thanks for all the help.

Things TODO:
1. Axis values are colliding with other figures
2. The x-axis values for Wasserstein div. figure (6) are colliding

**Pull request merged**